paddle/framework/shape_inference.h

```diff
@@ -24,6 +24,14 @@ class InferShapeContextBase {
   virtual ~InferShapeContextBase() {}
   virtual bool HasInput(const std::string &name) const = 0;
   virtual bool HasOutput(const std::string &name) const = 0;
+  bool HasInputs(const std::string &name) const {
+    const std::vector<std::string> &names = Inputs(name);
```

Comment:
I think we need to check if these variables are all in the scope~

Reply:
Done.